backend/app.py

```diff
@@ -99,8 +106,25 @@ def on_disconnect():
 
 @socketio.on('data')
 def handle_message(data):
+    """listener for data events"""
+    print("data from the front end: ", type(data))
+    emit("data", {'data': data, 'id': request.sid}, broadcast=True)
+
+
+@socketio.on('join')
```

Comment:
Oops, accidentally included some wip stuff I was working on with sockets. Should be ok I think. Will 100% be fixed before presenting on Monday in any case.